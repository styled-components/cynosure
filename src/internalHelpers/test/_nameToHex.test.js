// @flow

import nameToHex from '../_nameToHex'

describe('nameToHex', () => {
  it('should convert a named color to a hex value', () => {
    expect({ background: nameToHex('white') }).toEqual({
      background: '#fff',
    })
  })
  it('should convert a camel-cased named color to a hex value', () => {
    expect({ background: nameToHex('PowderBlue') }).toEqual({
      background: '#b0e0e6',
    })
  })
  it('should return a passed hex value without mutation', () => {
    expect({ background: nameToHex('#fff') }).toEqual({
      background: '#fff',
    })
  })
  it('should return a passed RGB string value without mutation', () => {
    expect({ background: nameToHex('rgb(0,0,0)') }).toEqual({
      background: 'rgb(0,0,0)',
    })
  })
  it('should return a passed HSL value without mutation', () => {
    expect({ background: nameToHex('hsl(180, 50%, 50%)') }).toEqual({
      background: 'hsl(180, 50%, 50%)',
    })
  })
  it('should return a non-string value without mutation', () => {
    // $FlowFixMe
    expect({ background: nameToHex(2) }).toEqual({
      background: 2,
    })
  })
})
