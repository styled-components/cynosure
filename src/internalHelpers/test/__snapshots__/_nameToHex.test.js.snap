exports[`nameToHex should convert a camel-cased named color to a hex value 1`] = `
Object {
  "background": "#b0e0e6",
}
`;

exports[`nameToHex should convert a named color to a hex value 1`] = `
Object {
  "background": "#fff",
}
`;

exports[`nameToHex should return a non-string value without mutation 1`] = `
Object {
  "background": 2,
}
`;

exports[`nameToHex should return a passed HSL value without mutation 1`] = `
Object {
  "background": "hsl(180, 50%, 50%)",
}
`;

exports[`nameToHex should return a passed RGB string value without mutation 1`] = `
Object {
  "background": "rgb(0,0,0)",
}
`;

exports[`nameToHex should return a passed hex value without mutation 1`] = `
Object {
  "background": "#fff",
}
`;
